actions/merkleroots/check_bhs_status.go

```diff
@@ -0,0 +1,47 @@
+package merkleroots
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/bitcoin-sv/spv-wallet/config"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog"
+)
+
+const (
+	errCheckingBHSMsg                  = "error checking Block Header Service"
+	errBlockHeadersServiceIsOfflineMsg = "Unable to connect to Block Headers Service. Please check Block Header Service configuration and status"
+)
+
+// checkBlockHeaderServiceStatus checks if the Block Header Service responds with 200 for a health check to see if it's
+// up and ready for connections
+func checkBlockHeaderServiceStatus(bhsConfig *config.BHSConfig, httpClient *resty.Client, logger *zerolog.Logger) bool {
+	logger.Info().Msg("checking Block Headers Service")
+
+	if bhsConfig.URL == "" {
+		logger.Error().Msgf("%s - url not configured", errCheckingBHSMsg)
+	}
+
+	if bhsConfig.AuthToken == "" {
+		logger.Warn().Msg("warning checking Block Headers Service - auth token is not set. Some requests might not work")
+	}
+
+	res, err := httpClient.
+		SetTimeout(5 * time.Second).
```

Comment:
If it's a global (for this http client) timeout - do it once where you make the client.

If it's a per-request do it after `R()`:

```golang 
resp, err := client.R().
    SetTimeout(3 * time.Second).
    Get("https://example.com")
```

Reply:
Urm, R(). doesn't have SetTimeout method, it's used per client, I actually forgot to remove it after this request. I will add that timeout removal afterwards. Unless you've got better idea how to tackle it

Reply:
I'm sorry I was suggested by ChatGPT... 
Indeed, there is no `SetTimeout` for single request.

I'm just thinking if it's a good idea to set the Timeout for the whole client.
I don't know if it's thread-safe.